distributed/worker.py

```diff
@@ -3182,6 +3193,25 @@ def parse_memory_limit(memory_limit, nthreads, total_cores=CPU_COUNT):
     return min(memory_limit, system.MEMORY_LIMIT)
 
 
+def parse_worker_ports(port):
+    if isinstance(port, str) and ":" not in port:
+        port = int(port)
+
+    if isinstance(port, (int, type(None))):
+        port = [port]
+    else:
+        port_start, port_stop = map(int, port.split(":"))
+        if port_stop <= port_start:
+            raise ValueError(
+                "When specifying a range of ports like port_start:port_stop, "
+                "port_stop must be greater than port_start, but got "
+                f"port_start={port_start} and port_stop={port_stop}"
+            )
+        port = list(range(port_start, port_stop + 1))
+
+    return port
```

Comment:
This could be used for nannies schedulers or other servers I think.  Maybe it should move to utils and we should remove `worker` from the name.  Also, can I ask for a small docstring, maybe with a couple of examples?

Should it always return a list of integers for consistency, or would that complicate other parts of the codebase?

Reply:
+1 to moving to utils and adding documentation

Right now this function either returns a list of integers or the 1-element list `[None]`. I decided to keep the `None` option as to not break existing code paths where if `None` is received as a port, then some configured [default port is used](https://github.com/dask/distributed/blob/ecdcb33720b2794c21b5bd322e4d2ccd8fec2787/distributed/core.py#L301-L302)